Snowflake.Data.Tests/SFConnectionIT.cs

```diff
@@ -744,6 +746,31 @@ public void TestSSOConnectionWithUser()
             }
         }
 
+        [Test]
+        [Ignore("This test requires manual interaction and therefore cannot be run in CI")]
+        public void TestSSOConnectionWithUserAsync()
+        {
+            using (SnowflakeDbConnection conn = new SnowflakeDbConnection())
+            {
+                conn.ConnectionString
+                    = ConnectionStringWithoutAuth
+                      + ";authenticator=externalbrowser;user=qa@snowflakecomputing.com";
+
+                Task connectTask = conn.OpenAsync(CancellationToken.None);
+                Assert.AreEqual(ConnectionState.Connecting, conn.State);

```

Comment:
Is there is no possibility of race here? If this line starts before async from the line above? Just curious :)

Reply:
If  Open connection happens quicker than expected this assert could be false cause connection will be already in Opened state, no?
Maybe we should remove this assert?